sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -321,4 +340,42 @@ case class InsertIntoHiveTable(
         isSrcLocal = false)
     }
   }
+
+  // scalastyle:off line.size.limit
+  /**
+   * If the table location and partition location do not belong to the same [[FileSystem]], We
+   * should not disable hive overwrite. Otherwise, hive will use the [[FileSystem]] instance belong
```

Comment:
can we treat this as a known limitation, with the faith that most users will use Hive 2.x in the long term? Then we just need to check the version.